### PR TITLE
Azure spring data cosmos version schema

### PR DESCRIPTION
**Pull request by saragluna**
The former design of Spring version schema will make two separate artifacts to cope with Spring Data 2.2.x and 2.3.x, but if we let Spring BOM manage our versions we could use one artifact to support both versions.

The dependency management section in a pom file could also affect the transitive versions: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#bill-of-materials-bom-poms.

### CODE REVIEW

**Review by kushagraThapar — APPROVED** [review]
Overall LGTM, 

@mitchdenny @danieljurek please review the pipeline changes.

File: sdk/cosmos/azure-spring-data-cosmos-test/pom.xml [hunk 1]
```diff
@@ -4,10 +4,10 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>com.azure</groupId>
-    <artifactId>azure-client-sdk-parent</artifactId>
-    <version>1.7.0</version> <!-- {x-version-update;com.azure:azure-client-sdk-parent;current} -->
-    <relativePath>../../parents/azure-client-sdk-parent</relativePath>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
```
Comment (mitchdenny) [inline sdk/cosmos/azure-spring-data-cosmos-test/pom.xml:8]: I'm not sure how updating this parent is going to impact some of the version bumping / nightly package logic.

**Pull request merged**